crates/cli/src/codegen/plugin.rs

```diff
@@ -1,147 +1,81 @@
-use crate::bytecode;
 use anyhow::{anyhow, bail, Result};
-use serde::Deserialize;
 use std::{
     fs,
-    io::{Read, Seek},
+    io::{self},
     path::Path,
     str,
 };
 use walrus::{ExportItem, ValType};
-use wasi_common::{pipe::WritePipe, sync::WasiCtxBuilder};
-use wasmtime::{AsContextMut, Engine, Linker};
 use wizer::Wizer;
 
-const PLUGIN_MODULE: &[u8] = include_bytes!(concat!(env!("OUT_DIR"), "/plugin.wasm"));
+use super::bytecode;
 
-/// Use the legacy plugin when using the `compile -d` command.
-const QUICKJS_PROVIDER_V2_MODULE: &[u8] = include_bytes!("./javy_quickjs_provider_v2.wasm");
-
-/// A property that is in the config schema returned by the plugin.
-#[derive(Debug, Deserialize)]
-#[serde(rename_all = "camelCase")]
-pub(crate) struct JsConfigProperty {
-    /// The name of the property (e.g., `simd-json-builtins`).
-    pub(crate) name: String,
-    /// The documentation to display for the property.
-    pub(crate) doc: String,
-}
-
-/// Different plugins that are available.
-#[derive(Debug)]
-pub enum Plugin {
-    /// The default plugin.
+/// Represents what kind of plugin.
+#[derive(Default, Clone)]
+pub(crate) enum PluginKind {
+    #[default]
+    None,
     Default,
-    /// A plugin for use with the `compile` to maintain backward compatibility.
     V2,
-    /// A plugin provided by the user.
-    User { bytes: Vec<u8> },
+}
+
+/// Represents any valid Javy plugin.
+#[derive(Clone, Debug)]
+pub struct Plugin {
+    pub bytes: Vec<u8>,
 }
 
 impl Default for Plugin {
     fn default() -> Self {
-        Self::Default
+        Self { bytes: Vec::new() }
     }
 }
 
 impl Plugin {
-    /// Creates a new user plugin.
-    pub fn new_user_plugin(path: &Path) -> Result<Self> {
-        Ok(Self::User {
-            bytes: fs::read(path)?,
-        })
+    /// Constructs a new instance of Plugin.
+    pub fn new(bytes: Vec<u8>) -> Self {
+        Plugin { bytes: bytes }
     }
 
-    /// Returns true if the plugin is a user plugin.
-    pub fn is_user_plugin(&self) -> bool {
-        matches!(&self, Plugin::User { .. })
-    }
-
-    /// Returns true if the plugin is the legacy v2 plugin.
-    pub fn is_v2_plugin(&self) -> bool {
-        matches!(&self, Plugin::V2)
+    /// Constructs a new instance of Plugin from a given path.
+    pub fn new_from_path<P: AsRef<Path>>(path: P) -> io::Result<Self> {
+        let bytes = fs::read(path)?;
+        Ok(Self::new(bytes))
     }
 
     /// Returns the plugin Wasm module as a byte slice.
     pub fn as_bytes(&self) -> &[u8] {
-        match self {
-            Self::Default => PLUGIN_MODULE,
-            Self::V2 => QUICKJS_PROVIDER_V2_MODULE,
-            Self::User { bytes } => bytes,
-        }
+        self.bytes.as_slice()
     }
+}
 
-    /// Uses the plugin to generate QuickJS bytecode.
+impl Plugin {
+    /// Uses a plugin to generate QuickJS bytecode.
     pub fn compile_source(&self, js_source_code: &[u8]) -> Result<Vec<u8>> {
```

Comment:
This probably doesn't matter quite yet but do we want/need this to be public? Do we not just want the `Generators` generate method to be public instead?

Reply:
Yeah a bunch of the methods that were public probably should be changed to `pub(super)` or `pub(crate)`.